src/software/thunderscope/thunderscope.py

```diff
@@ -247,34 +498,168 @@ def close(self):
+        action="store_true",
+        help="Run full system as the yellow team, over WiFi; estop required",
+    )
+    parser.add_argument(
+        "--interface",
+        action="store",
+        type=str,
+        default=None,
+        help="Which interface to communicate over",
+    )
 
-    elif args.run_simulator:
-        print(
-            "TODO #2050, this isn't implemented, just run the current standalone simulator"
-        )
+    # Sanity check that an interface was provided
+    args = parser.parse_args()
 
+    if args.run_blue or args.run_yellow:
+        if args.interface is None:
+            parser.error("Must specify interface")
+
+    tscope = Thunderscope()
+
+    ###########################################################################
+    #              AI + Robot Communication + Robot Diagnostics               #
+    ###########################################################################
+    #
+    # When we are running with real robots. We want to run 1 instance of AI
+    # and 1 instance of RobotCommunication which will send/recv packets over
+    # the provided multicast channel.
+    proto_unix_io = tscope.blue_full_system_proto_unix_io
+    runtime_dir = args.blue_fullsystem_runtime_dir
+    friendly_colour_yellow = False
+
+    if args.run_yellow:
+        proto_unix_io = tscope.yellow_full_system_proto_unix_io
+        runtime_dir = args.yellow_fullsystem_runtime_dir
+        friendly_colour_yellow = True
+
+    if args.run_blue or args.run_yellow:
+        # TODO (#2585): Support multiple channels
+        with RobotCommunication(
+            proto_unix_io, ROBOT_MULTICAST_CHANNEL_0, args.interface
+        ), FullSystem(
+            runtime_dir, args.debug_fullsystem, friendly_colour_yellow
+        ) as full_system:
+            tscope.load_saved_layout(
+                args.layout, load_blue=args.run_blue, load_yellow=args.run_yellow
+            )
+            tscope.show()
+
+    ###########################################################################
+    #           Blue AI vs Yellow AI + Simulator + Gamecontroller             #
+    ###########################################################################
+    #
+    # Run two AIs against each other with the er-force simulator. We also run
+    # the gamecontroller which can be accessed from http://localhost:8081
+    #
+    # The async sim ticket ticks the simulator at a fixed rate.
     else:
-        thunderscope = Thunderscope()
-        thunderscope.configure_default_layout()
-        thunderscope.show()
+
+        def __async_sim_ticker(tick_rate_ms):
+            """Setup the world and tick simulation forever
+
+            :param tick_rate_ms: The tick rate of the simulation
+
+            """
+            world_state = tbots_protobuf.create_world_state(
+                [geom.Point(3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)],
+                [geom.Point(-3, y) for y in numpy.linspace(-2, 2, NUM_ROBOTS)],
```

Comment:
doesn't seem like this would scale to the size of the field

Reply:
there is enough space between the robots that it does for both div A and B